cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/mapreduce/ETLMapReduce.java

```diff
@@ -134,8 +135,13 @@ public void configure() {
           .setTableProperty("avro.schema.literal", Constants.ERROR_SCHEMA.toString())
           .build(), true);
 
-    if (config.getResources() != null) {
-      setMapperResources(config.getResources());
+    Resources resources = config.getResources();
+    Resources driverResources = config.getDriverResources() == null ? resources : config.getDriverResources();
```

Comment:
So basically, driverResources defaults to mapper resources if driverResources is null?
Is that needed?
For instance, if we give 2 gigs of memory to mapper, and don't modify driver resources, driver will also get 2 gigs of memory. That doesn't seem necessary?